taichi/backends/metal/codegen_metal.cpp

```diff
@@ -1,4 +1,4 @@
-#include "taichi/codegen/codegen_metal.h"
+#include "codegen_metal.h"
```

Comment:
I'm not sure if I understand what you mean here.

Reply:
If you are talking about if we should use the full path when we `#include` taichi headers, we haven't reached a consensus yet, especially for **`#include "X.h"`/`#include "taichi/.../X.h"` in `X.cpp`**.

Maybe we should add **"always use full paths from the project root directory when including taichi headers"** as a convention to the code style? What do you think? @archibate @k-ye

Reply:
I think if you do sth like: rename cuda_driver.cpp to cuda/driver.cpp
Then this is important to make us concious about which header we want to include.

Reply:
> I think if you do sth like: `rename cuda_driver.cpp to cuda/driver.cpp`.
> Then this is important to make us concious about which header we want to include.

I'm still uncertain about what this means. I'm merging this PR now to prevent future merge conflicts. Feel free to open an issue to clarify your point. Thanks.